### PR TITLE
adding normalization, regrouped functions to increase readability

### PR DESCRIPTION
**Pull request by sedgewickmm18**
I added a set of new anomaly scorers with 'v2' appended to their names that "learn" and apply the proper scale of the input data before checking for anomalies

### CODE REVIEW

**Review by ankit-jha — APPROVED** [review]
What dataset are you using to test out the standard scaler approach ?

File: iotfunctions/anomaly.py [hunk 1]
```diff
@@ -1227,50 +1212,442 @@ def __init__(self, input_item, windowsize, output_item):
...
+            dfe = dfEntity
+
+        # interpolate gaps - data imputation
+        try:
+            dfe = dfe.interpolate(method="time")
+        except Exception as e:
+            logger.error('Prepare data error: ' + str(e))
+
+        # one dimensional time series - named temperature for catchyness
+        temperature = dfe[[self.input_item]].fillna(0).to_numpy().reshape(-1,)
```
Comment (singhshraddha) [inline iotfunctions/anomaly.py:1567]: Currently we are not applying the anomaly function on the scaled data, with the same random_state both this version and the old `KMeansAnomalyScore` return the same scores . `self.input_item` should be the replaced with the scaled/transformed input, which in our case is `'predicted_'+self.output.item` (but coded better)
```python3
temperature = dfe[['predicted_'+self.output.item]].fillna(0).to_numpy().reshape(-1,)
```

File: iotfunctions/anomaly.py [hunk 2]
```diff
@@ -1285,52 +1662,207 @@ def build_ui(cls):
...
```
Comment (singhshraddha) [inline iotfunctions/anomaly.py:1743]: same comment as the one in `KMeansAnomalyScorev2.prepare_data`
`self.input_item` is not the scaled_input, we are running the anomaly detection on the original data and not the scaled data

**Review by singhshraddha — COMMENTED** [review]
I wasn't able to run the new anomaly functions on local entity (which is a good debugging tool to have, and is consistent with previous anomaly functions). This is because `BaseEstimatorFunction` has two db operations that are not wrapped in a `if-else` or `try-except`. In base.py `BaseEstimatorFunction._execute`, if we wrap the two function calls 1. `self.evaluate_and_write_model` and 2. `self.get_models_for_predict` we can run these models with local entities that are not connected to any db

File: iotfunctions/anomaly.py (hunk 1 above)
Comment (sedgewickmm18) [reply iotfunctions/anomaly.py:1567]: good catch. Thank you very much for spotting this
I forgot to make sure I'm using the "predicted" (scaled) data for scoring after reorganizing the code.

File: iotfunctions/anomaly.py (hunk 2 above)
Comment (sedgewickmm18) [reply iotfunctions/anomaly.py:1743]: Same again.
prepare_data is now moved into Standard_Scaler and returns the predicted/scaled column for all subclasses.

**Comment by sedgewickmm18** [comment]
I've now also committed my notebook to test the scaling anomaly scorers, too. 
You'll find it [in my playground](https://github.com/sedgewickmm18/mmfunctions/blob/master/MonitorAnomalyFunctions.ipynb)
and it also shows how to use FileModelStore to run the scalers locally (expect a couple of files named 'KPI ...' to appear in your directory.)

**Review by singhshraddha — APPROVED** [review]

**Pull request merged**